chain-evm/src/machine.rs

```diff
@@ -70,6 +70,21 @@ pub type GasPrice = U256;
 /// Gas limit for EVM operations.
 pub type GasLimit = U256;
 
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+/// EVM Environment parameters needed for execution.
+pub struct Environment {
+    pub gas_price: GasPrice,
+    pub origin: Origin,
+    pub chain_id: ChainId,
+    pub block_hashes: BlockHashes,
```

Comment:
What is that supposed to be?

Reply:
This is a vec with a hash for each block in the chain, with the last item being the current block hash.

Same as with block_number, this needs to be updated everytime there is a new block.